engine/src/main/java/org/hibernate/validator/internal/engine/valuecontext/ValueContexts.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.valuecontext;
+
+import org.hibernate.validator.internal.engine.path.PathImpl;
+import org.hibernate.validator.internal.metadata.BeanMetaDataManager;
+import org.hibernate.validator.internal.metadata.aggregated.BeanMetaData;
+import org.hibernate.validator.internal.metadata.facets.Validatable;
+import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
+
+/**
+ * @author Marko Bekhta
+ */
+public final class ValueContexts<T, V> {
+
+	private ValueContexts() {
+	}
+
+	public static <T, V> ValueContextWithBeanMetadata<T, V> getLocalExecutionContext(
+			BeanMetaDataManager beanMetaDataManager,
+			ExecutableParameterNameProvider parameterNameProvider,
+			T value,
+			Validatable validatable,
+			PathImpl propertyPath) {
```

Comment:
I think we can remove this one and move the `beanMetaDataManager.getBeanMetaData( rootBeanType )` to the caller.

There's no real value in having one more method.

(It's not new but having all these almost identical variants always bothered me so let's simplify what we can)